osbs/build/build_requestv2.py

```diff
@@ -402,10 +447,14 @@ def set_data_from_reactor_config(self, reactor_config_data):
             if self.user_params.flatpak.value and not self.user_params.base_image.value:
                 raise OsbsValidationException(
                     "Flatpak base_image must be be set in container.yaml or reactor config")
-            else:
-                return
+        else:
+            self._set_flatpak(reactor_config_data)
 
-        self._set_flatpak(reactor_config_data)
+        if not self.source_registry:
+            raise RuntimeError('mandatory "source_registry" is not defined in reactor_config')
+        source_registry = RegistryURI(self.source_registry['url']).docker_uri
```

Comment:
Is it safe to always assume the source registry here? Why not take the pull registries into consideration?

Reply:
we need source registry to name imagestream and imagestream_tag for result of build

Reply:
Ah! Missed that this is setting up for the destination location. The source registry is the only possible destination registry.